src/panels/ActionsPanel/ActionsPanel.tsx

```diff
@@ -211,12 +244,21 @@ export default function ActionsPanel(): JSX.Element {
               cancelButtonLabel={Label.CANCEL_BUTTON}
               confirmButtonLabel={Label.CONFIRM_BUTTON}
               confirmButtonAppearance="positive"
-              onConfirm={() => {
+              onConfirm={(event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+                event.nativeEvent.stopImmediatePropagation();
                 setConfirmType(null);
-                executeAction().catch((error) =>
-                  console.error(Label.EXECUTE_ACTION_ERROR, error),
-                );
-                handleRemovePanelQueryParams();
+                executeAction()
+                  .then(() => {
+                    handleRemovePanelQueryParams();
+                    return;
+                  })
+                  .catch((error) => {
+                    setInlineErrors((prevInlineError) => [
```

Comment:
If you set the saving state when the action starts then you'll need to update it again here:

```suggestion
                  .catch((error) => {
                    setSaving(false);
                    setInlineErrors((prevInlineError) => [
```